overlord/snapstate/snapstate.go

```diff
@@ -69,25 +69,32 @@ func isParallelInstallable(snapsup *SnapSetup) error {
 	return fmt.Errorf("cannot install snap of type %v as %q", snapsup.Type, snapsup.InstanceName())
 }
 
+func canInstallSnapdSnap(st *state.State) error {
+	model, err := Model(st)
+	if err != nil && err != state.ErrNoState {
+		return err
+	}
+	if model == nil || model.Base() == "" {
+		tr := config.NewTransaction(st)
+		experimentalAllowSnapd, err := config.GetFeatureFlag(tr, features.SnapdSnap)
+		if err != nil && !config.IsNoOption(err) {
+			return err
+		}
+		if !experimentalAllowSnapd {
+			return fmt.Errorf("cannot install snapd snap on a model without a base snap yet")
+		}
```

Comment:
Hmm, I'm slightly confused, is this logic correct? Shouldn't we check `experimentalAllowSnapd` before even checking the model and error out with "cannot install ... because the experimental snapd-snap feature is disabled"; then missing base is a separate error? As is, we will transition everyone as long as there is a base snap, regardless of the flag. Did I miss something?

Reply:
Yeah, this looks a bit flimsy I think this needs some careful checking. I will do that.